ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/tools/TopicInfoCache.java

```diff
@@ -102,7 +102,16 @@ public TopicInfoCache(
   }
 
   public TopicInfo get(final String topicName) {
-    return cache.getUnchecked(topicName);
+    final TopicInfo topicInfo = cache.getUnchecked(topicName);
+
+    // Value formats can not be determined if a null (tombstone) value was produced in the topic.
+    // If the valueFormat is not determined, then attempt to refresh the entry cached to get the
+    // right format in case a non-null value is already produced.
+    if (topicInfo.getValueFormat().getFormat().equals(FormatFactory.NONE.name())) {
+      cache.refresh(topicName);
```

Comment:
For my own education: what is this `cache` thing? And won't we need to get `TopicInfo` from the cache again, after we refreshed it? What does `refresh()` do internally (ie, how does a refresh work)?

Reply:
The `cache.getUnchecked(topicName)` will call the `load(final String topicName)` internally if no items exist in the cache. After that is called, consecutive calls to `getUnchecked` will use the item in the cache, and `load` will not be called any more until the item has expired. 

The problem I am fixing is that sometimes, the first record found in the shared outer changelog topic has a null value. This causes the `getTopicInfo` to return a `NO_VALUE_SCHEMA_FOUND`. When that info is obtained from the `load` function, then it is kept in the cache for some time. I know that the null value is just something that may not happen too often, so a new record in the chanelog topic should allow the TrackSerde to detect the value format. However, the cache will not have the new format until the item expires, so in cases of the NONE value found, I refresh the cache so it triggers the `load` method again in case the value format was detected.

Reply:
Is that the only time that it can be `None`? Otherwise we risk always refreshing the cache and getting the same result (I think so, but worth double checking)